### PR TITLE
I2C HAL Updates

### PR DESCRIPTION
**Pull request by CatGarab**
Updating the C I2C HAL to not use the old FreeRTOS-based style (and also yotta).

Updating the various APIs to correctly use the new HAL updates

### CODE REVIEW

**Comment by CatGarab** [comment]
[Update C I2C HAL to be more Linux-y](https://trello.com/c/NRFQdG3F/240-update-c-i2c-hal-to-be-more-linux-y)

File: hal/kubos-hal/kubos-hal/i2c.h [hunk 1]
```diff
@@ -154,117 +52,41 @@ typedef enum {
...
+ * This fuction is used to terminate an active I2C bus connection.
+ * It takes a pointer to the file descriptor to be closed.
+ * After calling this function the device will *not* be available for usage in the reading/writing functions.
  *
  * Example usage:
  * @code
 // initialize bus
-k_i2c_init(K_I2C1, &conf);
+int bus;
+k_i2c_init("/dev/i2c-1", bus);
```
Comment (plauche) [inline hal/kubos-hal/kubos-hal/i2c.h]: nit: Should be `k_i2c_init("/dev/i2c-1", &bus);`

File: hal/kubos-hal/kubos-hal/i2c.h [hunk 2]
```diff
@@ -154,117 +52,41 @@ typedef enum {
...
  *
  * Example usage:
  * @code
- KI2CConf conf = {
-     .addressing_mode = K_ADDRESSING_MODE_7BIT,
-     .role = K_MASTER,
-     .clock_speed = 100000
- };
- k_i2c_init(K_I2C1, &conf);
+ k_i2c_init("/dev/i2c-1");
```
Comment (plauche) [inline hal/kubos-hal/kubos-hal/i2c.h]: nit: Needs reference to some `int bus`

File: hal/kubos-hal/source/i2c.c [hunk 3]
```diff
@@ -15,100 +15,93 @@
...
+#include <unistd.h>
 
-static KI2C k_i2cs[K_NUM_I2CS];
-
-KI2CStatus k_i2c_init(KI2CNum i2c, KI2CConf *conf)
+KI2CStatus k_i2c_init(char * device, int * fp)
 {
-    KI2C *k_i2c = kprv_i2c_get(i2c);
-    if ((k_i2c != NULL) && (k_i2c->bus_num == K_I2C_NO_BUS))
+
```
Comment (plauche) [inline hal/kubos-hal/source/i2c.c]: Nit: Null check `device` and `fp`

File: hal/kubos-hal/source/i2c.c [hunk 4]
```diff
@@ -15,100 +15,93 @@
...
 
     return I2C_OK;
 }
 
-void k_i2c_terminate(KI2CNum i2c)
+void k_i2c_terminate(int * fp)
 {
-    KI2C *k_i2c = kprv_i2c_get(i2c);
-    if ((k_i2c != NULL) && (k_i2c->bus_num != K_I2C_NO_BUS))
+    if (*fp == 0)
```
Comment (plauche) [inline hal/kubos-hal/source/i2c.c]: Nit: Null check `fp`

**Pull request merged**